extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -1236,6 +1250,58 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
       (returnedSizeDifferent && heightNotIncreased)
     ) {
       this.attemptChangeSize(newHeight, newWidth).catch(() => {});
+      if (newWidth > width) {
+        this.adjustSlotPostExpansion(newWidth);
+      }
+    }
+  }
+
+  /**
+   * Ensures that slot is properly centered after being expanded.
+   * @param {number} newWidth The new width of the slot.
+   */
+  adjustSlotPostExpansion(newWidth) {
+    if (this.flexibleAdSlotData_ == null) {
+      dev().warn(
+        TAG,
+        'Attempted to expand slot without flexible ad slot data.'
+      );
+      return;
+    }
+    const isRtl = isRTL(this.win.document);
+    const dirStr = isRtl ? 'Right' : 'Left';
+    // Guaranteed to be set after exiting if/else.
+    let /** ?number */ newMargin = null;
+    if (newWidth <= this.flexibleAdSlotData_.parentWidth) {
+      // Must center creative within its parent container
+      const {parentStyle} = this.flexibleAdSlotData_;
+      const parentPadding = parseInt(parentStyle[`padding${dirStr}`], 10) || 0;
+      const parentBorder =
+        parseInt(parentStyle[`border${dirStr}Width`], 10) || 0;
+      const whitespace = (this.flexibleAdSlotData_.parentWidth - newWidth) / 2;
+      newMargin = whitespace - parentPadding - parentBorder;
+    } else {
+      // Must center creative within the viewport
```

Comment:
I'm not sure I understand why that is?

Reply:
Say the viewport width is 400px, and the body has 15px padding. If we expand the slot to 400px, it'll get cut off by 15px on the right unless we center it.